cs/src/Connections/ITunnelHost.cs

```diff
@@ -78,6 +89,15 @@ public interface ITunnelHost : IAsyncDisposable
             TunnelPort updatedPort,
             CancellationToken cancellation);
 
-        // TODO: Events about tunnel port and connection activity?
+        /// <summary>
+        /// Event handler for refreshing the tunnel access token.
+        /// The tunnel client will fire this event when it is not able to use the access token it got from the tunnel.
+        /// </summary>
+        EventHandler<RefreshingTunnelAccessTokenEventArgs>? RefreshingTunnelAccessToken { get; set; }
```

Comment:
Having an event handler as a get/set property will be confusing. Switching back to an `event` is fine.